src/Bicep.Core/Workspaces/SourceFileFactory.cs

```diff
@@ -23,29 +24,32 @@ public static class SourceFileFactory
             LineInfoHandling = LineInfoHandling.Ignore,
         };
 
-        public static BicepSourceFile? TryCreateSourceFileByBicepLanguageId(Uri fileUri, string fileContents, string languageId) => languageId switch
+
+        public static ISourceFile CreateSourceFile(Uri fileUri, string fileContents, Type? sourceFileType = null)
```

Comment:
>Type [](http://example.com/codeflow?start=85&length=4)

Should this be an enum instead to limit the options that can be passed in? Or if we want to keep it as `Type`, should we validate that it's either null or one of the types that we expect?

Reply:
I considered adding SourceFileKind, but we already have BicepSourceFileKind, which applies only to BicepFile and BicepParamFile. Merging BicepSourceFileKind into SourceFileKind might be a good idea in the future, but for now, I'll add validation for `sourceFileType`.

Reply:
Sounds good